File: demo-app/android/observe-sdk/src/main/java/com/observe/sdk/core/ObserveSession.kt
package com.observe.sdk.core

import android.os.Build
import java.util.UUID

/**
 * Represents a single observe recording session
 */
data class ObserveSession(
    val id: String,
    val startTime: Long,
    val appVersion: String,
    val platform: String,
    val deviceInfo: DeviceInfo
) {
    companion object {
        fun create(appVersion: String, platform: String): ObserveSession {
            return ObserveSession(
                id = UUID.randomUUID().toString(),
                startTime = System.currentTimeMillis(),
                appVersion = appVersion,
                platform = platform,
                deviceInfo = DeviceInfo.current()
            )
        }
    }
}

data class DeviceInfo(
    val model: String,
    val manufacturer: String,
    val osVersion: String,
    val sdkInt: Int
) {
    companion object {
        fun current(): DeviceInfo {
            return DeviceInfo(
                model = Build.MODEL,
                manufacturer = Build.MANUFACTURER,
                osVersion = Build.VERSION.RELEASE,
                sdkInt = Build.VERSION.SDK_INT
            )
        }
    }
}

File: demo-app/ios/ObserveSDK/Core/ObserveSession.swift
//
//  ObserveSession.swift
//  ObserveSDK
//
//  Represents an observation session
//

import Foundation
import UIKit

/// Represents a single observation session
public struct ObserveSession: Codable {
    
    /// Unique session identifier
    public let sessionId: String
    
    /// Session start timestamp
    public let startTime: Int64
    
    /// App version
    public let appVersion: String
    
    /// Platform (always "ios")
    public let platform: String
    
    /// Device information
    public let deviceInfo: DeviceInfo
    
    /// Number of events exported in this session
    public var eventsExported: Int
    
    // MARK: - Nested Types
    
    public struct DeviceInfo: Codable {
        public let model: String
        public let osVersion: String
        public let screenWidth: Int
        public let screenHeight: Int
        public let locale: String
        
        public init(
            model: String,
            osVersion: String,
            screenWidth: Int,
            screenHeight: Int,
            locale: String
        ) {
            self.model = model
            self.osVersion = osVersion
            self.screenWidth = screenWidth
            self.screenHeight = screenHeight
            self.locale = locale
        }
    }
    
    // MARK: - Initializer
    
    public init(
        sessionId: String,
        startTime: Int64,
        appVersion: String,
        platform: String,
        deviceInfo: DeviceInfo,
        eventsExported: Int = 0
    ) {
        self.sessionId = sessionId
        self.startTime = startTime
        self.appVersion = appVersion
        self.platform = platform
        self.deviceInfo = deviceInfo
        self.eventsExported = eventsExported
    }
    
    // MARK: - Factory
    
    /// Create a new session with current device info
    public static func create(appVersion: String = "1.0.0") -> ObserveSession {
        let device = UIDevice.current
        let screen = UIScreen.main.bounds
        
        let deviceInfo = DeviceInfo(
            model: device.model,
            osVersion: device.systemVersion,
            screenWidth: Int(screen.width),
            screenHeight: Int(screen.height),
            locale: Locale.current.identifier
        )
        
        return ObserveSession(
            sessionId: UUID().uuidString,
            startTime: Int64(Date().timeIntervalSince1970 * 1000),
            appVersion: appVersion,
            platform: "ios",
            deviceInfo: deviceInfo,
            eventsExported: 0
        )
    }
}

